src/Cmdlets/GetKubeResourceCmdlet.cs

```diff
@@ -11,22 +11,52 @@ namespace Kubectl.Cmdlets {
     [Cmdlet(VerbsCommon.Get, "KubeResource")]
     [OutputType(new[] { typeof(KubeResourceV1) })]
     public sealed class GetKubeResourceCmdlet : KubeApiCmdlet {
+        private const string DefaultParamSet = "DefaultParamSet";
+        private const string NamespaceObjectSet = "NamespaceObjectSet";
+
         [Parameter(ValueFromPipelineByPropertyName = true)]
         [Alias("Ns")]
         public string Namespace { get; set; } = "default";
 
-        [Parameter(Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true)]
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = NamespaceObjectSet,
+            ValueFromPipeline = true)]
+        [ValidateNotNull()]
+        public NamespaceV1 NamespaceObject { get; set; }
```

Comment:
This is an interesting case. This code gets (the latest version of) a deployment:
```powershell
Get-KubeDeployment -Name some-deploy | Get-KubeResource
```
With this code
```powershell
Get-KubeNamespace -Name some-ns | Get-KubeResource
```
what would you expect it to get: the _namespace_ itself or _resources inside the namespace_? After this PR it would do the latter, but before this PR it was the former

Reply:
It should be the resources inside of the namespace. I didn't test this condition enough, if it should be that way, I'll make a test case to make sure this doesn't break.